pkg/virtual/workspaces/auth/watch.go

```diff
@@ -0,0 +1,252 @@
+package auth
+
+import (
+	"errors"
+	"sync"
+
+	"k8s.io/klog/v2"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/apiserver/pkg/authentication/user"
+	kstorage "k8s.io/apiserver/pkg/storage"
+
+	"github.com/kcp-dev/kcp/pkg/apis/tenancy/projection"
+	workspaceapi "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
+	workspaceapibeta1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1beta1"
+	workspacecache "github.com/kcp-dev/kcp/pkg/virtual/workspaces/cache"
+)
+
+type CacheWatcher interface {
```

Comment:
is this whole file coming from openshift? Can you add a reference here back to the original in a comment?

Reply:
This is the case for the whole `auth` package and there is already a comment in the package `doc.go` file.

Reply:
ack